### ISSUE TITLE
Coloring the route line based on height/slope does not work

### ISSUE THREAD

**Issue by Nine-Tailed** [osmandapp/OsmAnd#19624]
### Description

Coloring the route line based on height/slope does not work.
<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/12549744/ef1e15c9-06b7-452e-8fa3-0f908f421838"  width="300" height="auto"/> </td>
  </tr>
</table>

### Steps to reproduce

Customize the coloring of the route line for navigation depending on the height or slope.

### Actual result

Coloring depending on height/slope does not work.

### Expected result

Coloring based on height/slope should work.

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version: OsmAnd~ 4.8.0#2146m, Дата выпуска: 2024-04-19
Android/iOS version: 13
Device model: poco f3
Crash-Logs: ?
```

**Comment by sonora** [comment]
Yes, I can confirm this. A regression likely introduced with the new appearance config UI

See also https://github.com/osmandapp/OsmAnd/issues/19625 in this context.

**Comment by Nine-Tailed** [comment]
Why does route coloring not work in Khibiny when using the Brouter router?
See video in telegram chat
https://t.me/ruosmand/150534?single
Khibiny mountains are here: https://www.openstreetmap.org/node/6483117776

**Comment by Nine-Tailed** [comment]
![Screenshot_2024-04-27-22-00-18-959_net.osmand.plus.jpg](https://github.com/osmandapp/OsmAnd/assets/12549744/037825d0-f5e3-4621-b2a0-71dc24e51b69)

Something was broken again in the coloring of the route line, the correct coloring flashes for a split second and then turns dark blue.

**Comment by sonora** [comment]
Can't reproduce this time round - at the moment (4.7.12) it seems to work OK for me...

**Comment by Nine-Tailed** [comment]
@sonora on render v2?

**Comment by sonora** [comment]
Ah, true, no prob in V1, but reproducible in V2.

**Comment by Nine-Tailed** [comment]
possibly connected, the transparency of the track has also disappeared, it is drawn dark blue.
https://github.com/osmandapp/OsmAnd/issues/19702

**Comment by Guylby** [comment]
As of 4.7.13 Android, coloring the line based on height/slope works but all other modes revert to default track color:

https://github.com/osmandapp/OsmAnd/assets/151167525/dfd6590f-9215-4b28-acc0-e07c82ca94ad

**Comment by DmitryAlexei** [comment]
@Guylby Cannot reproduce your issue using regular navigation. 
Or do you use gpx track for navigation? Please try regular navigation and also try to attach your track to roads

**Comment by Guylby** [comment]
I've created a new profile based on the default Osmand foot navigation profile.

I started a route calculation between two randoms points
The route is shown
Then I go to the profile options and Customize route line option. I try the different options but only "Custom" and "Altitude" work correctly (can't test Slope as I don't have Osmand Pro), all other track rendering styles just show the default map style color:
Even with no data they should show a grey path as the tooltip suggests

https://github.com/osmandapp/OsmAnd/assets/151167525/4861a5ee-419b-4c7a-b821-a4989c94b248

Changing to the default "Osmand" map rendering style still shows the same behaviour.
